app/src/main/java/com/openlattice/chronicle/services/status/EnrollmentStatusMonitoringService.kt

```diff
@@ -103,24 +103,28 @@ class EnrollmentStatusMonitoringService : JobService() {
 
             // schedule notifications for active questionnaires
             for ((key, value) in studyQuestionnaires) {
-                val recurrenceRule = value[FullQualifiedName(RECURRENCE_RULE)]?.iterator()?.next()?.toString()
+                val recurrenceRuleSet = value[FullQualifiedName(RECURRENCE_RULE)]?.iterator()?.next()?.toString()
                 val name = value[FullQualifiedName(NAME)]?.iterator()?.next()?.toString()
                 val active = value[FullQualifiedName(ACTIVE)]?.iterator()?.next()?.equals(true)
 
-                if (!recurrenceRule.isNullOrEmpty() && !name.isNullOrEmpty()) {
-                    notification = NotificationEntry(
-                            key.toString(),
-                            NotificationType.QUESTIONNAIRE,
-                            recurrenceRule,
-                            name,
-                            "Tap to complete questionnaire"
-                    )
-
-                    intent = Intent(this, NotificationsService::class.java).apply {
-                        putExtra(NOTIFICATION_ENTRY,  Gson().toJson(notification))
-                        putExtra(NOTIFICATIONS_ENABLED, active != null && active && participationStatus == ParticipationStatus.ENROLLED)
+                if (!recurrenceRuleSet.isNullOrEmpty() && !name.isNullOrEmpty()) {
+
+                    recurrenceRuleSet.split("RRULE:").filter { it.isNotEmpty() }.forEach {
+                        run {
```

Comment:
What does the `run` block do? You didn't have it before, why do you have it now?

Reply:
alternative for
```
forEach{
  notification = ...
}
```
It's unnecessary though, so I have discarded it.